### PR TITLE
fix #118 - also catch IOExceptions mentioning ChannelClosedException

### LINKED ISSUE #118: GCE workers sometimes die
```
Cannot contact runner-01-gce: java.io.IOException: remote file operation failed: /var/lib/jenkins/dlang_projects@5/BlackEdder/ggplotd at hudson.remoting.Channel@2ad430c7:JNLP4-connect connection from 30.0.195.35.bc.googleusercontent.com/35.195.0.30:58308: hudson.remoting.ChannelClosedException: Remote call on JNLP4-connect connection from 30.0.195.35.bc.googleusercontent.com/35.195.0.30:58308 failed. The channel is closing down or has closed down

script returned exit code -1
```

https://ci.dlang.io/blue/organizations/jenkins/dlang-org%2Fdmd/detail/PR-7661/3/pipeline

### PR DESCRIPTION
**Pull request by MartinNowak**

### CODE REVIEW

File: vars/runPipeline.groovy [hunk 1]
```diff
@@ -103,9 +103,16 @@ def retryOnPreemption (action, times = 1) {
         try {
```
Comment (wilzbach) [inline vars/runPipeline.groovy]: Don't you want to set `times` higher than `1`?
If I understand it correctly it wouldn't do the restart even if the exception is properly.

File: vars/runPipeline.groovy [hunk 2]
```diff
@@ -103,9 +103,16 @@ def retryOnPreemption (action, times = 1) {
...
             action()
             return
-        // Jenkins looses contact to executor
+        // Jenkins might loose contact to preemptible executors
```
Comment (JinShil) [inline vars/runPipeline.groovy]: nit: lose

File: vars/runPipeline.groovy [hunk 3]
```diff
@@ -103,9 +103,16 @@ def retryOnPreemption (action, times = 1) {
...
         } catch (ClosedChannelException e) {
-            echo "${e.toString()}"
+            echo e.toString()
+        // Nested exception catched somewhere in Channel object (see #118)
```
Comment (JinShil) [inline vars/runPipeline.groovy]: nit: caught

File: vars/runPipeline.groovy (hunk 2 above)
Comment (MartinNowak) [reply vars/runPipeline.groovy]: There is an edit button for nits ;).

File: vars/runPipeline.groovy (hunk 1 above)
Comment (MartinNowak) [reply vars/runPipeline.groovy]: renamed to `retries` for clarification

Comment (MartinNowak) [reply vars/runPipeline.groovy]: The last attempt rethrows https://github.com/dlang/ci/pull/124/files#diff-e15b7462050c1530715a62b67a9b7e02R118.

Comment (wilzbach) [reply vars/runPipeline.groovy]: Ah okay. I would still increase it it 2 or 3 as I have seen these exceptions rather frequently in the last weeks.

Comment (MartinNowak) [reply vars/runPipeline.groovy]: Yeah, despite opposite claims, GCE is quite aggressively preempting instances. Maybe I need to switch datacenters.

**Pull request merged**